xbstrap/__init__.py

```diff
@@ -22,11 +22,14 @@
 
 main_parser = argparse.ArgumentParser()
 main_parser.add_argument("-v", dest="verbose", action="store_true", help="verbose")
+main_parser.add_argument(
+    "-S", type=str, dest="source_dir", help="source dir (in place of bootstrap.link)"
+)
 main_subparsers = main_parser.add_subparsers(dest="command")
 
 
 def do_runtool(args):
-    cfg = xbstrap.base.config_for_dir()
+    cfg = xbstrap.base.config_for_dir(args.source_dir)
```

Comment:
Since this pattern is used so frequently in this file, we should add a `config_from_args` helper function at the top of the file. That will also potentially allow us to override the build dir etc. later on.